### PR TITLE
Added function for CesiumRTC

### PR DESCRIPTION
**Pull request by lasalvavida**
@lilleyse

### CODE REVIEW

**Comment by lilleyse** [comment]
Should the positions be transformed relative to the center point too?

File: lib/addCesiumRTC.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+var Ellipsoid = Cesium.Ellipsoid;
+var defined = Cesium.defined;
+
+var addExtensionsUsed = require('./addExtensionsUsed');
+
+module.exports = addCesiumRTC;
+
+function addCesiumRTC(gltf, options) {
+    var positionArray = [];
+    var position = Cartesian3.fromRadians(options.longitude, options.latitude, options.height, Ellipsoid.WGS84);
```
Comment (lilleyse) [inline lib/addCesiumRTC.js]: Seems easier to just compute the center from the position attribute min/max. It's a little more restrictive, but this makes it easier to add this stage to `gltfPipeline`.

**Comment by lilleyse** [comment]
As we talked about offline, I had a misunderstanding of the RTC extension. My model is specified in wgs84 already so I need to do some preprocessing before using this stage.

File: lib/addCesiumRTC.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * @param {Number} [options.latitude] The latitude to map onto the ellipsoid.
+ * @param {Number} [options.height] Distance from the ellipsoid's surface.
+ * @param {Ellipsoid} [options.ellipsoid=Ellipsoid.WGS84] The ellipsoid to use for lat-long mapping.
+ */
+function addCesiumRTC(gltf, options) {
+    if (defined(options)) {
+        var positionArray = [];
+        var ellipsoid = defaultValue(options.ellipsoid, Ellipsoid.WGS84);
+        var position = defaultValue(options.position,
+            Cartesian3.fromRadians(options.longitude, options.latitude, options.height, ellipsoid));
```
Comment (lilleyse) [inline lib/addCesiumRTC.js]: If `options.position` is defined it will still call `Cartesian3.fromRadians` but with undefined values.

Comment (lasalvavida) [inline lib/addCesiumRTC.js]: Yup, good catch

**Comment by lasalvavida** [comment]
Updated

**Comment by lilleyse** [comment]
Looks good.

File: lib/addCesiumRTC.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Positions the model relative to a center point using the CESIUM_RTC extension.
+ * Either specify options.position directly or calculate it from longitude, latitude and height
+ * along with an ellipsoid (WGS84 if not specified).
+ *
+ * @param {Object} gltf A javascript object holding a glTF hierarchy.
+ * @param {Object} [options] Defines the relationship with the center position
+ * @param {Cartesian3} [options.position] Places the model relative to a particular position.
+ * @param {Number} [options.longitude] The longitude to map onto the ellipsoid.
```
Comment (pjcozzi) [inline lib/addCesiumRTC.js]: We are usually explicit that longitude/latitude is in radians and height is in meters.

File: lib/addCesiumRTC.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+function addCesiumRTC(gltf, options) {
+    if (defined(options)) {
+        var positionArray = [];
+        var ellipsoid = defaultValue(options.ellipsoid, Ellipsoid.WGS84);
+        var position = options.position;
+        if (!defined(position)) {
+            if (defined(options.longitude) && defined(options.latitude && defined(options.height))) {
+                position = Cartesian3.fromRadians(options.longitude, options.latitude, options.height, ellipsoid);
+            } else {
+                throw new DeveloperError('Either a position or lat/long/height must be provided');
```
Comment (pjcozzi) [inline lib/addCesiumRTC.js]: Don't abbreviate the parameters here.

**Comment by pjcozzi** [comment]
> As we talked about offline, I had a misunderstanding of the RTC extension. My model is specified in wgs84 already so I need to do some preprocessing before using this stage.

Exactly, but I don't know that this stage is useful in practice here.  Although it is appropriate for our 3D Tiles pipeline; it is too niche for the glTF Pipeline, and just increases its complexity with a stage that most users will not know what to do with.

I suggest that we remove it.

**Comment by lasalvavida** [comment]
> I suggest that we remove it.

I'm not sure that's the best avenue. This is a glTF extension and I think that having this utility could be useful. Maybe just make the documentation a little clearer on what it is for?

**Comment by pjcozzi** [comment]
This stage alone is not useful since it doesn't update the vertex data.  It's not a priority for us to add that to gltf-pipeline now and most users would not need it, so I would remove this partial stage and reconsider adding a complete stage if users need it in practice.

**Pull request merged**